modules/resource-discovery/main.tf

```diff
@@ -0,0 +1,7 @@
+resource "aws_vpc_ipam_resource_discovery" "this" {
+  description = var.description
+  operating_regions {
```

Comment:
This block can almost certainly be specified multiple times to enable multiple regions. To model that, we can use a list of `operating_regions` as the input variable, and the [dynamic block feature](https://developer.hashicorp.com/terraform/language/expressions/dynamic-blocks).